collector-sqs/src/main/java/zipkin/collector/sqs/AwsSqsCollector.java

```diff
@@ -0,0 +1,202 @@
+
+    @Override public CollectorComponent.Builder metrics(CollectorMetrics metrics) {
+      delegate.metrics(Util.checkNotNull(metrics, "metrics").forTransport("sqs"));
+      return this;
+    }
+
+    @Override public CollectorComponent.Builder sampler(CollectorSampler sampler) {
+      delegate.sampler(sampler);
+      return this;
+    }
+
+    /** SQS queue URL to consume from */
+    public Builder queueUrl(String queueUrl) {
+      this.queueUrl = queueUrl;
+      return this;
+    }
+
+    /** AWS credentials for authenticating calls to SQS. */
+    public Builder credentialsProvider(AWSCredentialsProvider credentialsProvider) {
+      this.credentialsProvider = credentialsProvider;
+      return this;
+    }
+
+    /** Amount of time to wait for messages from SQS */
+    public Builder waitTimeSeconds(int seconds) {
+      checkArgument(parallelism > 0 && parallelism < 21, "parallelism");
+      this.waitTimeSeconds = seconds;
+      return this;
+    }
+
+    /** How many processors to run in parallel for each queue URL */
+    public Builder parallelism(int parallelism) {
+      checkArgument(parallelism > 0, "parallelism");
+      this.parallelism = parallelism;
+      return this;
+    }
+
+    @Override public CollectorComponent build() {
+      return new AwsSqsCollector(this);
+    }
+
+    Builder() {
+    }
+  }
+
+  private final LazyAmazonSQSAsync client;
+  private final LazyProcessors processors;
+
+  AwsSqsCollector(Builder builder) {
+    client = new LazyAmazonSQSAsync(builder.credentialsProvider);
+    processors = new LazyProcessors(client.get(), builder.delegate.build(), builder.queueUrl,
+        builder.parallelism, builder.waitTimeSeconds);
+  }
+
+  @Override public CollectorComponent start() {
+    processors.get();
+    return this;
+  }
+
+  @Override public CheckResult check() {
+    try {
+      client.get(); // make sure compute doesn't throw an exception
+      return processors.check(); // check if any processor has failed
+    } catch (RuntimeException e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override public void close() throws IOException {
+    client.close();
+    processors.close();
+  }
+
+  private static final class LazyProcessors extends LazyCloseable<List<AwsSqsSpanProcessor>>
+      implements Component {
+
+    private static final Logger logger = Logger.getLogger(LazyProcessors.class.getName());
+
+    final String queueUrl;
+    final int parallelism;
+    final AmazonSQSAsync client;
+    final Collector collector;
+    final int waitTimeSeconds;
+
+    LazyProcessors(AmazonSQSAsync client, Collector collector, String queueUrl, int parallelism, int waitTimeSeconds) {
+      this.queueUrl = queueUrl;
+      this.parallelism = parallelism;
+      this.client = client;
+      this.collector = collector;
+      this.waitTimeSeconds = waitTimeSeconds;
+    }
+
+    @Override protected List<AwsSqsSpanProcessor> compute() {
+      return IntStream.range(0, parallelism - 1)
```

Comment:
careful. unless you need java 8, this might limit adoption. ex retrolambda can backport programming style, but not libraries like IntStream. I think the highest lang level of other collector components in 7